### PR TITLE
fix new function spec

### PR DESCRIPTION
**Pull request by ghost**
spec should allow function call with one argument.

### CODE REVIEW

File: lib/ecto_job/job_queue.ex [hunk 1]
```diff
@@ -146,7 +146,7 @@ defmodule EctoJob.JobQueue do
        - `:schedule` : runs the job at the given `%DateTime{}`
        - `:max_attempts` : the maximum attempts for this job
       """
-      @spec new(map, Keyword.t()) :: EctoJob.JobQueue.job()
+      @spec new(map, Keyword.t() | nil) :: EctoJob.JobQueue.job()
```
Comment (mbuhot) [inline lib/ecto_job/job_queue.ex:149]: the call to `Keyword.get` on line 154 requires the second argument to be a `Keyword.t`.

Are you getting a dialyzer error that adding `| nil` fixes?

Comment (ghost) [reply lib/ecto_job/job_queue.ex:149]: Yes it's for dialyzer when calling new/1

Comment (mbuhot) [reply lib/ecto_job/job_queue.ex:149]: Please post the error here. I'm pretty sure `| nil` isn't correct for this function.

Comment (ghost) [reply lib/ecto_job/job_queue.ex:149]: Yes nil is not a correct parameter but the second one should be optional in the spec.

```
apps/app/lib/job_queue.ex:5:invalid_contract
Invalid type specification for function.

Function:
App.JobQueue.new/2

Success typing:
@spec new(map(), Keyword.t()) :: %App.JobQueue{
  :__meta__ => %Ecto.Schema.Metadata{
    :context => nil,
    :prefix => nil,
    :schema => App.JobQueue,
    :source => <<_::64>>,
    :state => :built
  },
  :attempt => 0,
  :expires => nil,
  :id => nil,
  :inserted_at => nil,
  :max_attempts => _,
  :notify => _,
  :params => map(),
  :schedule => _,
  :state => <<_::72>>,
  :updated_at => nil
}
```

And the call is:  `job = JobQueue.new(%{account_id: account_id, task: "check"})`

Comment (mbuhot) [reply lib/ecto_job/job_queue.ex:149]: I was able to reproduce this issue in the demo project.
Dialyzer stopped complaining once I added the `__meta__` key to the `job` type.

I'll close this PR for now. If the latest master commit doesn't fix the problem in your app, please open an issue.

**Pull request closed without merging**